### ISSUE TITLE
Is it possible to iterate over each node of an onnx model?

### ISSUE THREAD

**Issue by sreenikSS** [Microsoft/onnxruntime#787]
I want to build a converter for pretrained onnx models to another framework not yet supported by onnx (mlpack). What can be a suitable way to get started so that for each layer I obtain the layer type and then iterate over the nodes accessing their weights and biases?

**Comment by faxu** [comment]
See this example: https://github.com/onnx/onnx/blob/master/onnx/examples/load_model.ipynb

**Comment by raymondxyang** [comment]
IMO it's more like an onnx issue. see above example for general model loading. But it only showed the shape info (value_info) for the tensors. Weights are stored in initializer field at(https://github.com/onnx/onnx/blob/master/onnx/onnx.in.proto#L258). You can load the ModelProto from onnx.onnx_pb sub-module and access the fields listed in the .proto file.

I'll close the issue for now and please feel free to follow up with any remaining questions here or in onnx/onnx repo.

**Comment by sreenikSS** [comment]
Thanks a lot. Yes I had actually figured it out shortly after. I should have left a comment. 

I have another question though, using the C++ bindings is it possible to create an onnx model? I had found very few issues related to the C++ bindings in the onnx/onnx repo and sadly this isn't answered anywhere. 
To clarify a bit, say I need to build either a `linear-relu-linear-softmax` or a `linear-relu-linear-sigmoid` model depending on the user's input, is it supported in C++?

This being a small doubt I don't actually want to open a new issue in onnx/onnx, hence I am asking this in the comments here.
